BlockchainSdk/Blockchains/Sui/SuiError.swift

```diff
@@ -0,0 +1,16 @@
+//
+// SuiError.swift
+// BlockchainSdk
+//
+// Created by Sergei Iakovlev on 30.08.2024
+// Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+
+struct SuiError {
+    enum CoinType: Error {
+        case failedEncoding
+        case failedDecoding
+    }
+}
```

Comment:
гораздо лучше отражает о чем речь.
```suggestion
struct SuiError {
    enum CodingError: Error {
        case failedEncoding
        case failedDecoding
    }
}
```

Но в целом можно и не вкладывать в структуру, а сразу, если нет каких-то ещё планов с добавлением ошибок

```suggestion
enum SuiError: Error {
    case failedEncoding
    case failedDecoding
}
```